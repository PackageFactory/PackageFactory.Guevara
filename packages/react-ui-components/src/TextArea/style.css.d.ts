export interface IStyleCss {
  'textArea': string;
  'textArea--invalid': string;
  'textArea--highlight': string;
  'textArea--disabled': string;
}

export const locals: IStyleCss;
