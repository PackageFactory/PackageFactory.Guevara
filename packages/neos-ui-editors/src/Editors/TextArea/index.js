import React from 'react';
import PropTypes from 'prop-types';
import TextArea from '@neos-project/react-ui-components/src/TextArea/';

const defaultOptions = {
    disabled: false,
    maxlength: null,
    readonly: false,
    placeholder: '',
    minRows: 2,
    expandedRows: 6
};

const TextAreaEditor = props => {
    const {id, value, commit, options, className} = props;

    const finalOptions = Object.assign({}, defaultOptions, options);

    return (<TextArea
        id={id}
        value={value}
        className={className}
        onChange={commit}
        disabled={finalOptions.disabled}
        maxLength={finalOptions.maxlength}
        readOnly={finalOptions.readonly}
        placeholder={finalOptions.placeholder}
        minRows={finalOptions.minRows}
        expandedRows={finalOptions.expandedRows}
        />
    );
};
TextAreaEditor.propTypes = {
    id: PropTypes.string,
    value: PropTypes.string,
    highlight: PropTypes.bool,
    commit: PropTypes.func.isRequired,
    options: PropTypes.object,
    validationErrors: PropTypes.array
};

export default TextAreaEditor;
