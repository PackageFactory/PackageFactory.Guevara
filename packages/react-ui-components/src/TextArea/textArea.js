import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import TextareaAutoresize from 'react-textarea-autosize';
import enhanceWithClickOutside from 'react-click-outside';

class TextArea extends PureComponent {
    state = {
        isFocused: false
    };

    static propTypes = {
        /**
         * An optional className to render on the textarea node.
         */
        className: PropTypes.string,

        /**
         * An optional HTML5 placeholder.
         */
        placeholder: PropTypes.string,

        /**
         * The handler which will be called once the user changes the value of the input.
         */
        onChange: PropTypes.func,

        /**
         * This prop controls if the CheckBox is disabled or not.
         */
        disabled: PropTypes.bool,

        /**
         * An optional css theme to be injected.
         */
        theme: PropTypes.shape({
            textArea: PropTypes.string,
            'textArea--invalid': PropTypes.string
        }).isRequired,

        /**
         * Optional number to set the minRows of the TextArea if not expanded
         */
        minRows: PropTypes.number,

        /**
         * Optional number to set the expandedRows of the TextArea if expanded
         */
        expandedRows: PropTypes.number
    };

    static defaultProps = {
        minRows: 2,
        expandedRows: 6
    };

    handleValueChange = e => {
        const value = e.target.value;
        const {onChange} = this.props;

        if (onChange) {
            onChange(value);
        }
    }

    handleOnClick = () => {
        this.setState({
            isFocused: true
        });
    }

    handleClickOutside = () => {
        this.setState({
            isFocused: false
        });
    }

    render() {
        const {
            placeholder,
            className,
            validationErrors,
            theme,
            highlight,
            disabled,
            minRows,
            expandedRows,
            ...rest
        } = this.props;
        const classNames = mergeClassNames({
            [className]: className && className.length,
            [theme.textArea]: true,
            [theme['textArea--disabled']]: disabled
        });

        return (
            <div>
                <TextareaAutoresize
                    {...rest}
                    className={classNames}
                    role="textbox"
                    aria-multiline="true"
                    aria-disabled={disabled ? 'true' : 'false'}
                    placeholder={placeholder}
                    disabled={disabled}
                    onChange={this.handleValueChange}
                    onClick={this.handleOnClick}
                    minRows={this.state.isFocused ? expandedRows : minRows}
                    />
            </div>
        );
    }
}

//
// Add the click-outside functionality to the TextArea component.
//
const EnhancedTextArea = enhanceWithClickOutside(TextArea);

export default EnhancedTextArea;
export const undecorated = TextArea;
