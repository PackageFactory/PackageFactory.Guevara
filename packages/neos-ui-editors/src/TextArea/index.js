import React from 'react';
import PropTypes from 'prop-types';
import TextArea from '@neos-project/react-ui-components/src/TextArea/';

const TextAreaEditor = props => {
    const {value, commit, highlight, options} = props;

    const disabled = options && options.disabled ? options.disabled : false;
    const maxlength = options && options.maxlength ? options.maxlength : null;
    const readonly = options && options.readonly ? options.readonly : false;

    return (<TextArea
        value={value}
        onChange={commit}
        highlight={highlight}
        disabled={disabled}
        maxLength={maxlength}
        readOnly={readonly}
        />
    );
};
TextAreaEditor.propTypes = {
    value: PropTypes.string,
    highlight: PropTypes.bool,
    commit: PropTypes.func.isRequired,
    options: PropTypes.object
};

export default TextAreaEditor;
