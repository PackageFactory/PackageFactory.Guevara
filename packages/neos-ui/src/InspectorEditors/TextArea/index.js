import React, {PropTypes} from '@host/react';
import TextArea from '@neos-project/react-ui-components/lib/TextArea/';

const TextAreaEditor = props => {
    const {value, commit} = props;

    return <TextArea value={value} onChange={commit}/>;
};
TextAreaEditor.propTypes = {
    value: PropTypes.string.isRequired,
    commit: PropTypes.func.isRequired
};

export default TextAreaEditor;
