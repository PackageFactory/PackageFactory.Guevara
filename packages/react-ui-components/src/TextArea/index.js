import {themr} from 'react-css-themr';
import identifiers from './../identifiers.js';
import style from './style.css';
import TextArea from './textArea.js';

const ThemedTextArea = themr(identifiers.textArea, style)(TextArea);

export default ThemedTextArea;
