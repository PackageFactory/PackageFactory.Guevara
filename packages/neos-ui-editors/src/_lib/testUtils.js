import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
