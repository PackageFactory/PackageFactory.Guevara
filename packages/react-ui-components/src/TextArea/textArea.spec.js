import sinon from 'sinon';
import TextareaAutoresize from 'react-textarea-autosize';
import {createShallowRenderer} from './../_lib/testUtils.js';

import TextArea from './textArea.js';

const defaultProps = {
    theme: {}
};
const shallow = createShallowRenderer(TextArea, defaultProps);

test('should render a "TextareaAutoresize" component.', () => {
    const input = shallow().find(TextareaAutoresize);

    expect(input.length).toBe(1);
});
test('should call the passed "onChange" prop with the value of the input when changing it.', () => {
    const onChange = sinon.spy();
    const input = shallow({onChange}).find(TextareaAutoresize);

    input.simulate('change', {
        target: {
            value: 'my value'
        }
    });

    expect(onChange.callCount).toBe(1);
    expect(onChange.args[0][0]).toBe('my value');
});
