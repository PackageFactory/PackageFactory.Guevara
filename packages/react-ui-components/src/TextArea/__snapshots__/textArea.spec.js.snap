// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea/> should render correctly. 1`] = `
<div>
  <TextareaAutosize
    aria-disabled="false"
    aria-multiline="true"
    className=""
    minRows={2}
    onChange={[Function]}
    onClick={[Function]}
    onHeightChange={[Function]}
    role="textbox"
    useCacheForDOMMeasurements={false}
  />
</div>
`;
